jaxrs-api/src/main/java/javax/ws/rs/core/Response.java

```diff
@@ -791,6 +791,21 @@ public static ResponseBuilder temporaryRedirect(URI location) {
         return status(Status.TEMPORARY_REDIRECT).location(location);
     }
 
+     /**
+     * Create a new ResponseBuilder for a permanent redirection.
+     *
+     * @param location the redirection URI. If a relative URI is
+     *                 supplied it will be converted into an absolute URI by resolving it
+     *                 relative to the base URI of the application (see
+     *                 {@link UriInfo#getBaseUri}).
+     * @return a new response builder.
+     * @throws java.lang.IllegalArgumentException
+     *          if location is {@code null}.
+     */
+    public static ResponseBuilder permanentRedirect(URI location) {
```

Comment:
I'm not convinced that a shortcut method `permanentRedirect` is required. A lot of other, IMO more important, status codes don't have shortcut methods either. I would rather not bloat this class with another method -- if we decide to add the enum.

Reply:
I share this opinion.

Reply:
Well, these shortcut methods are especially important for status codes that need additional headers to be set. Playing devil's advocate here, actually 308 would be a good example for a status method for which a shortcut method would make sense.

Reply:
@mkarg @sdaschner : I suggest for now, lets keep it as is ...ie to have the shortcut method.

Reply:
We should only consider adding this method if a majority agrees on its usefulness. And currently most comments questioned it. So let's wait for other comments.

Reply:
I don't have a problem keeping the method.

Reply:
@spericas Does that mean "plus one" or "zero" from your side? Just for correctly counting votes.

Reply:
@sdaschner "Minus one" due to the shortcut method, or "zero" because it was only a comment? Just for counting votes correctly.

Reply:
Zero then. When I think about it more, I do see the advantage that you want to add an URI, similar to `created`, what @chkal mentioned.

Reply:
@mkarg It is a +1 for me. This error code is widely supported in all major browsers, that is good enough for me. As I said, I also don't have a problem keeping the static method.

Reply:
Sum so far (as not everybody used the Github review tool): chkal 0, mkarg -1, spericas +1, sdaschner 0 => result 0.